homeassistant/components/google_assistant/trait.py

```diff
@@ -2394,18 +2410,62 @@ def supported(cls, domain, features, device_class, _):
     def sync_attributes(self):
         """Return attributes for a sync request."""
         device_class = self.state.attributes.get(ATTR_DEVICE_CLASS)
-        if (data := self.sensor_types.get(device_class)) is not None:
+        data = self.sensor_types.get(device_class)
+
+        if device_class is None or data is None:
+            return {}
+        
+        if device_class == sensor.SensorDeviceClass.AQI:
             return {
-                "sensorStatesSupported": {
-                    "name": data[0],
-                    "numericCapabilities": {"rawValueUnit": data[1]},
-                }
+                "sensorStatesSupported": [
+                    {
+                        "name": data[0],
+                        "numericCapabilities": {"rawValueUnit": data[1]},
+                        "descriptiveCapabilities": {
+                            "availableStates": [
+                                "healthy",
+                                "moderate",
+                                "unhealthy for sensitive groups",
+                                "unhealthy",
+                                "very unhealthy",
+                                "hazardous",
+                                "unknown",
+                            ],
+                        },
+                    }
+                ]
+            }
+        if data is not None:
```

Comment:
Data can't be none here. Also it looks much of the data is common between the two cases?

Looks to only differ by descriptice capability. Would make more sense to just conditipn that on the class.

Reply:
Great point! I was moving too fast. I consolidated some of the logic and updated the tests. Should be much better now!